vm/rust/src/jsonrpc.rs

```diff
@@ -190,6 +195,53 @@ impl From<BlockifierOrderedEvent> for OrderedEvent {
     }
 }
 
+#[derive(Serialize)]
+pub struct ExecutionResources {
+    pub steps: usize,
+    pub memory_holes: usize,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub range_check_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pedersen_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub poseidon_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ec_op_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ecdsa_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bitwise_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keccak_builtin_applications: Option<usize>,
+    // https://github.com/starkware-libs/starknet-specs/pull/167
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub segment_arena_builtin: Option<usize>,
+}
+
+use cairo_vm::vm::runners::cairo_runner::ExecutionResources as VmExecutionResources;
+impl From<VmExecutionResources> for ExecutionResources {
+    fn from(val: VmExecutionResources) -> Self {
+        let applications = |builtin_name: &str| -> Option<usize> {
+            match val.builtin_instance_counter.get(builtin_name) {
+                Some(x) => Some(*x),
+                None => None,
+            }
+        };
+        ExecutionResources {
+            steps: val.n_steps,
+            memory_holes: val.n_memory_holes,
+            range_check_builtin_applications: applications(RANGE_CHECK_BUILTIN_NAME),
```

Comment:
you can do `val.builtin_instance_counter.get(RANGE_CHECK_BUILTIN_NAME).cloned()` if you feel like it looks cleaner